### ISSUE TITLE
`prefer-svelte-reactivity` reports in non-runes components, where `SvelteSet` & co. are not reactive

### ISSUE THREAD

**Issue by badreddinesaadioui** [sveltejs/eslint-plugin-svelte#1580]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.x

### What version of `eslint-plugin-svelte` are you using?

3.23.0 (svelte 5.57.1)

### What did you do?

A legacy-mode component (no runes) that keeps a plain `Set` as a non-reactive cache:

```svelte
<script>
	const seen = new Set();
	let count = 0;

	function visit(id) {
		if (seen.has(id)) return;
		seen.add(id);
		count += 1;
	}
</script>

<button on:click={() => visit('a')}>{count}</button>
```

### What did you expect to happen?

No report. This component is in legacy mode, and following the suggestion does not make anything reactive there.

### What actually happened?

```
2:18  error  Found a mutable instance of the built-in Set class. Use SvelteSet instead  svelte/prefer-svelte-reactivity
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/badreddinesaadioui/prefer-svelte-reactivity-legacy-repro

`npx eslint Lint.svelte` shows the report, and `node --conditions=browser run.mjs` shows `SvelteSet` not updating in the legacy component.

#### Why the suggestion does not apply in legacy mode

In a non-runes component the compiler does not track `svelte/reactivity` classes. Compiling a legacy component that reads a `SvelteSet` in the template and in a `$:` statement gives:

```js
$.legacy_pre_effect(() => {}, () => {
	$.set(viaDollar, s.size);
});
// ...
$.template_effect(() => {
	$.set_text(text, ($.untrack(() => s.size)));
	// ...
});
```

The template read is wrapped in `$.untrack` and the `$:` statement has no dependencies, so `s.add(...)` updates neither. At runtime (`run.mjs` in the repro), after two `add()` calls the legacy component still renders `0` in both places, while the same component in runes mode renders `2`.

So in legacy mode the rule points you at a fix that does nothing, and the only way to silence it is to rewrite working code (I ended up replacing a `Set` with an array).

#### Proposed fix

Gate the rule on runes mode, the same way `prefer-writable-derived` already does:

```ts
conditions: [
	{
		svelteVersions: ['5'],
		svelteFileTypes: ['.svelte', '.svelte.[js|ts]'],
		runes: [true, 'undetermined']
	}
]
```

`.svelte.js/.ts` files are always runes, so they are unaffected. Components that are runes through a rune, `<svelte:options runes />` or `compilerOptions.runes` are still reported.

This is not the same as #1271: that issue covers runes components where the instance never escapes a function. This one covers legacy components, where the suggested classes are never reactive at all.

I have the fix and tests ready and can open a PR.
